griptape/tools/google_drive/tool.py

```diff
@@ -0,0 +1,335 @@
+from __future__ import annotations
+import logging
+from schema import Schema, Literal, Optional, Or
+from attr import define, field
+from griptape.artifacts import ErrorArtifact, InfoArtifact, ListArtifact, BlobArtifact, TextArtifact
+from griptape.utils.decorators import activity
+from griptape.tools import BaseGoogleClient, BaseTool
+from google.auth.exceptions import MalformedError
+from googleapiclient.discovery import Resource
+from googleapiclient.http import MediaIoBaseUpload
+from googleapiclient.errors import HttpError
+from io import BytesIO
+
+
+@define
+class GoogleDriveClient(BaseGoogleClient, BaseTool):
+    LIST_FILES_SCOPES = ['https://www.googleapis.com/auth/drive.readonly']
+    UPLOAD_FILE_SCOPES = ['https://www.googleapis.com/auth/drive.file']
+    GOOGLE_EXPORT_MIME_MAPPING = {
+        "application/vnd.google-apps.document":
+            "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+        "application/vnd.google-apps.spreadsheet":
+            "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+        "application/vnd.google-apps.presentation":
+            "application/vnd.openxmlformats-officedocument.presentationml.presentation"
+    }
+
+    owner_email: str = field(kw_only=True)
+
+    @activity(config={
+        "description": "Can be used to list files in a specific Google Drive folder or the root directory",
+        "schema": Schema({
+            Literal("folder_path",
+                    description="Path of the Google Drive folder (like 'MainFolder/Subfolder1/Subfolder2') "
+                                "from which files should be listed. Specify 'root' to list files from the "
```

Comment:
Once it's made optional, you don't need to prompt the LLM to provide "root" as a value.

Reply:
Done!